pkg/controller/controller_utils.go

```diff
@@ -54,6 +54,13 @@ var (
 	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type ResyncPeriodFunc func() time.Duration
```

Comment:
I see, now it makes sense, thank you.

On Wed, Oct 7, 2015 at 1:08 PM, Wojciech Tyczynski <notifications@github.com
> wrote:

> In pkg/controller/controller_utils.go
> <https://github.com/kubernetes/kubernetes/pull/15153#discussion_r41439676>
> :
>
> > @@ -54,6 +54,13 @@ var (
> >  	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
> >  )
> >
> > +type ResyncPeriodFunc func() time.Duration
>
> It can call the function more than once, because it can have more than one
> Informer (one informer per resource). I wanted those also to not collide
> (e.g. daemon controller is using this function for both pods and nodes).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15153/files#r41439676>.
>